src/main/java/com/shopping_app/shoppingApp/config/ProblemAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,26 @@
+package com.shopping_app.shoppingApp.config;
+
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class ProblemAuthenticationEntryPoint implements AuthenticationEntryPoint {
```

Comment:
What purpose does this class serve?

Reply:
yeah if we want to handle auth exception and also to return proper exception message  to client(UI) we need to add this class